lighthouse-core/gather/gatherers/trace-elements.js

```diff
@@ -93,18 +96,26 @@ class TraceElements extends Gatherer {
           RectHelpers.getRectArea(newRect) -
           RectHelpers.getRectOverlapArea(oldRect, newRect);
 
-        let prevShiftTotal = 0;
-        if (clsPerNodeMap.has(node.node_id)) {
-          prevShiftTotal += clsPerNodeMap.get(node.node_id);
-        }
-        clsPerNodeMap.set(node.node_id, prevShiftTotal + areaOfImpact);
-        clsNodeIds.add(node.node_id);
+        pixelsMovedPerNode.set(node.node_id, areaOfImpact);
+
+        totalAreaOfImpact += areaOfImpact;
+      });
+
+      [...pixelsMovedPerNode.entries()].forEach(entry => {
```

Comment:
```
for (const [key, pixelsMoved] of pixelsMovedPerNode.entries()) { ...
```

for-loops are better than `.forEach`, and destructuring key-value entry into variables is nice for readability